src/components/App.svelte

```diff
@@ -9,19 +9,21 @@ import Footer from './Footer.svelte'
 import routes from '../views/routes'
 import Bootstrap from './Bootstrap.svelte'
 import Error from './Error.svelte'
+import { get } from 'svelte/store'
 
-const publicRoutes = ['/terms', '/privacy']
+const publicRoutes = ['/login', '/terms', '/privacy']
 
 $: isPublicRoute = publicRoutes.some(publicRoute => $location.startsWith(publicRoute))
-$: isProtectedRoute = ! isPublicRoute
+$: ! (isPublicRoute || isUserAuthn()) && authenticate() // should only react to location changes, not user changes.
 $: userIsAuthn = $me.id
-$: userIsNotAuthn = ! userIsAuthn
-$: isProtectedRoute && userIsNotAuthn && authenticate()
-$: minimal = $location.startsWith('/welcome') || userIsNotAuthn
+$: minimal = $location.startsWith('/welcome') || ! userIsAuthn
 $: isDataNeeded = userIsAuthn && ! minimal
 $: isDataNeeded && loadData()
 $: $location && scrollTo(0,0) // ensure route changes behave like a normal page change in the browser by going back to the top of the page.
 
+function isUserAuthn() {
+  return get(me).id
```

Comment:
Small preference
```suggestion
  return $me.id
```

Reply:
agreed, I like this better!  https://github.com/silinternational/wecarry-ui/pull/129/commits/6eff05732fc916edfa6b45fb5e00956f650a24aa